app/models/data.py

```diff
@@ -42,18 +51,48 @@ class Batch(db.Model, DataMixin):
 
     coreData = relationship('CoreData', backref='batch')
 
+    # This method isn't used when the object is read from the DB; only when a new one is being
+    # created, as from a POST JSON payload.
     def __init__(self, **kwargs):
-        super(Batch, self).__init__(**kwargs)
+        # parse datetime fields
+
+        # if there is no createdAt field, set it to datetime now
+        if 'createdAt' not in kwargs:
+            kwargs['createdAt'] = pytz.utc.localize(datetime.now())
+        else:
+            logging.info(
+                'New batch came in with existing createdAt: %s' % kwargs['createdAt'])
+
+        # setting default values for isPublished, isRevision: mimics preview state
+        kwargs['isPublished'] = False
+        kwargs['isRevision'] = False
+
+        mapper = class_mapper(Batch)
+        relevant_kwargs = {k: v for k, v in kwargs.items() if k in mapper.attrs.keys()}
+        super(Batch, self).__init__(**relevant_kwargs)
+
+
+    def to_dict(self):
+        d = super(Batch, self).to_dict()
+        d['coreData'] = [coreData.to_dict() for coreData in self.coreData]
+        return d
 
 
 class State(db.Model, DataMixin):
     __tablename__ = 'states'
 
     state = db.Column(db.String, primary_key=True, nullable=False)
-    fullName = db.Column(db.String)
+    name = db.Column(db.String)
+    covid19Site = db.Column(db.String)
+    covid19SiteOld = db.Column(db.String)
+    covid19SiteSecondary = db.Column(db.String)
+    twitter = db.Column(db.String)
+    notes = db.Column(db.String)
```

Comment:
Do we need to include the deprecated fields like `pui` and `pum` here? Even if they're not really useful, the public API still includes them.